kratos/geometries/geometry_data.h

```diff
@@ -791,61 +735,21 @@ class GeometryData
 
     virtual void save( Serializer& rSerializer ) const
     {
-        rSerializer.save( "Dimension", mDimension );
-        rSerializer.save( "Working Space Dimension", mWorkingSpaceDimension );
-        rSerializer.save( "Local Space Dimension", mLocalSpaceDimension );
-//    rSerializer.save("Default Method", mDefaultMethod);
-//    rSerializer.save("Integration Points", mIntegrationPoints);
-//    rSerializer.save("Shape Functions Values", mShapeFunctionsValues);
-//    rSerializer.save("Shape Functions Local Gradients", mShapeFunctionsLocalGradients);
+        rSerializer.save("GeometryDimension", mpGeometryDimension);
+        rSerializer.save("GeometryShapeFunctionContainer", mGeometryShapeFunctionContainer);
     }
 
     virtual void load( Serializer& rSerializer )
     {
-        rSerializer.load( "Dimension", mDimension );
-        rSerializer.load( "Working Space Dimension", mWorkingSpaceDimension );
-        rSerializer.load( "Local Space Dimension", mLocalSpaceDimension );
-//    rSerializer.load("Default Method", mDefaultMethod);
-//    rSerializer.load("Integration Points", mIntegrationPoints);
-//    rSerializer.load("Shape Functions Values", mShapeFunctionsValues);
-//    rSerializer.load("Shape Functions Local Gradients", mShapeFunctionsLocalGradients);
+        rSerializer.load("GeometryDimension", const_cast<GeometryDimension*>(mpGeometryDimension));
+        rSerializer.load("GeometryShapeFunctionContainer", mGeometryShapeFunctionContainer);
```

Comment:
Is there any test for this? Seems that you are creating a new Goemetry dimension which is not the static one!

Reply:
Actually I have made a test. However, it can be that this test always creates a new one here and does not keep the static one. This, I don't know how to test.

Reply:
You may check the address in the new pointer with the address of the static one to see if it points to the same static one.